tests/basic/lazy_static.rs

```diff
@@ -0,0 +1,226 @@
+use shuttle::scheduler::DfsScheduler;
+use shuttle::thread::ThreadId;
+use shuttle::{check_dfs, check_random, thread, Runner};
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use test_log::test;
+
+#[test]
+fn basic() {
+    shuttle::lazy_static! {
+        static ref HASH_MAP: HashMap<u32, u32> = {
+            let mut m = HashMap::new();
+            m.insert(1, 1);
+            m
+        };
+    }
+
+    check_dfs(|| assert_eq!(HASH_MAP.get(&1), Some(&1)), None);
+}
+
+#[test]
+fn racing_init() {
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    static COUNTER: AtomicUsize = AtomicUsize::new(0);
+
+    shuttle::lazy_static! {
+        static ref HASH_MAP: HashMap<u32, u32> = {
+            let mut m = HashMap::new();
+            m.insert(1, 1);
+            COUNTER.fetch_add(1, Ordering::SeqCst);
+            m
+        };
+    }
+
+    check_dfs(
+        || {
+            let thds = (0..2)
+                .map(|_| {
+                    thread::spawn(move || {
+                        assert_eq!(HASH_MAP.get(&1), Some(&1));
+                    })
+                })
+                .collect::<Vec<_>>();
+
+            for thd in thds {
+                thd.join().unwrap();
+            }
+
+            assert_eq!(COUNTER.swap(0, Ordering::SeqCst), 1);
+        },
+        None,
+    );
+}
+
+#[test]
+fn init_with_yield() {
+    shuttle::lazy_static! {
+        static ref THING: Arc<usize> = {
+            // check that it's valid to yield in the initializer
+            thread::yield_now();
+            Default::default()
+        };
+    }
+
+    check_dfs(
+        || {
+            let thd = thread::spawn(|| {
+                assert_eq!(**THING, 0);
+            });
+
+            assert_eq!(**THING, 0);
+
+            thd.join().unwrap();
+        },
+        None,
+    );
+}
+
+#[test]
+fn mutex_dfs() {
```

Comment:
Could you improve the test name, and add a comment -- this test checks that DFS will explore the N! interleavings during initialization?

Reply:
Done, and ended up refactoring a little to share the two tests.